text/src/autogluon/text/automm/predictor.py

```diff
@@ -467,33 +439,125 @@ def fit(
             teacher_model, critics, baseline_funcs, soft_label_loss_func,\
                 teacher_df_preprocessor, teacher_data_processors = None, None, None, None, None, None
 
-        self._fit(
+        _fit_args = dict(
             train_df=train_data,
             val_df=tuning_data,
-            df_preprocessor=df_preprocessor,
-            data_processors=data_processors,
-            model=model,
-            config=config,
-            loss_func=loss_func,
             validation_metric=validation_metric,
             validation_metric_name=validation_metric_name,
             custom_metric_func=custom_metric_func,
             minmax_mode=minmax_mode,
-            teacher_model=teacher_model,
-            critics=critics,
-            baseline_funcs=baseline_funcs,
-            soft_label_loss_func=soft_label_loss_func,
-            teacher_df_preprocessor=teacher_df_preprocessor,
-            teacher_data_processors=teacher_data_processors,
             max_time=time_limit,
-            save_path=save_path,
-            ckpt_path=self._ckpt_path,
-            resume=self._resume,
-            enable_progress_bar=self._enable_progress_bar,
-            mixup_fn=self.mixup_fn,
+            save_path=os.path.abspath(save_path) if hyperparameter_tune_kwargs is not None else save_path,
+            ckpt_path=None if hyperparameter_tune_kwargs is not None else self._ckpt_path,
+            resume=False if hyperparameter_tune_kwargs is not None else self._resume,
+            enable_progress_bar=False if hyperparameter_tune_kwargs is not None else self._enable_progress_bar,
+            config=config,
+            hyperparameters=hyperparameters,
+            teacher_predictor=None if hyperparameter_tune_kwargs is not None else teacher_predictor,
```

Comment:
May mark a TODO here that we won't support HPO if we are using teacher predictor.

Reply:
Or even raise a warning.

Reply:
Why can't HPO support distillation?